lib/chef/resource/locale.rb

```diff
@@ -25,67 +25,115 @@ class Locale < Chef::Resource
       description "Use the locale resource to set the system's locale."
       introduced "14.5"
 
+      LC_VARIABLES = %w{LC_ADDRESS LC_COLLATE LC_CTYPE LC_IDENTIFICATION LC_MEASUREMENT LC_MESSAGES LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER LC_TELEPHONE LC_TIME}.freeze
+      LOCALE_CONF = "/etc/locale.conf".freeze
+      LOCALE_REGEX = /\A\S+/.freeze
+
       property :lang, String,
-               default: "en_US.utf8",
-               description: "Sets the default system language."
+               description: "Sets the default system language.",
+               regex: [LOCALE_REGEX],
+               validation_message: "The provided lang not valid. It should be a non-empty string without any leading whitespaces."
+
+      property :lc_env, Hash,
+               description: "A Hash of LC_* env variables in the form of ({'LC_ENV_VARIABLE' => 'VALUE'}).",
```

Comment:
Can we make this example Ruby 1.9+ hash format

Reply:
That works for _symbolic_ keys, we are expecting and working on _string_ keys.
We can convert keys in string in coerce if its not going to be an overhead. Could you please share your views.